db/dcrpg/pgblockchain.go

```diff
@@ -628,8 +628,8 @@ func (pgb *ChainDB) VersionCheck(client *rpcclient.Client) error {
 		return nil
 	}
 
-	// AddAuxPendingDBUpgrades adds the pending db upgrades and reindexes.
-	_, err := pgb.AddAuxPendingDBUpgrades(client, needsUpgrade[0].CurrentVer,
+	// UpgradeTables adds the pending db upgrades and reindexes.
+	_, err := pgb.UpgradeTables(client, needsUpgrade[0].CurrentVer,
```

Comment:
The ordering issue is here too.

Reply:
On this one I might use some more clarification. `needsUpgrade` only holds table version that require `Upgrade` or `Reindex`. So if the code got here then an upgrade or reindexing must happen.

What could possibly be affected is where that upgrade starts from but from our earlier discussion only a newly created table(with upgrade type `OK`) has different current and required table version since the rest have similar table versions.

Unless how we fetch the table versions is buggy (which could make it harder to determine the current and required version) IMO I think this should be okay.

Reply:
Yes, you are correct.  Thanks for going over that.